pdk/1.2/modules/ROOT/pages/policies-pdk-configure-features-cors.adoc

```diff
@@ -0,0 +1,254 @@
+
+CORS is a mechanism for web applications to access resources defined in another domain. Browsers support CORS by default. The PDK CORS library complies with the https://fetch.spec.whatwg.org/[CORS W3C recommendation^] standards and provides a set of tools to validate incoming requests and protect resources.
+
+[[cors-configuration]]
+== Define a CORS Configuration Instance
+
+The CORS library is available from the `pdk::cors` module. The module provides the `cors::Configuration`, `cors::OriginGroup`, and `cors::AllowedMethod` structs and their builders to define how to validate incoming requests. For example, the following code snippet defines a function that hardcodes `cors::Configuration` instance:
+
+[source,Rust]
+----
+use pdk::cors;
+
+fn create_cors_configuration() -> cors:Configuration {
+    cors::Configuration::builder()
+        .public_resource(false)
+        .support_credentials(true)
+        .origin_groups(vec![cors::OriginGroup::builder()
+            .origin_group_name("default group".to_string())
+            .plain_origins(vec!["http://www.the-origin-of-time.com".to_string()])
+            .access_control_max_age(60)
+            .headers(vec![
+                "x-allow-origin".to_string(),
+                "x-yet-another-valid-header".to_string(),
+            ])
+            .exposed_headers(vec!["x-forwarded-for".to_string()])
+            .allowed_methods(vec![
+                cors::AllowedMethod::builder()
+                    .method_name("POST".to_string())
+                    .allowed(true)
+                    .build()
+                    .unwrap(),
+                cors::AllowedMethod::builder()
+                    .method_name("PUT".to_string())
+                    .allowed(true)
+                    .build()
+                    .unwrap(),
+            ])
+            .build()
+            .unwrap()])
+        .build()
+        .unwrap();
+}
+----
+
+[[configuration-methods]]
+=== cors::Configuration Struct Configuration Parameters
+
+[%header%autowidth.spread,cols="a,a,a,a"]
+|===
+|Property | Required or Optional | Default Value | Description
+
+| `origin_groups`
+| Optional
+| Empty `Vec`
+| `Vec` containing groups of origins configurable with `cors::OriginGroup`.
+
+| `public_resource`
+| Optional
+| `false`
+| A boolean indicating if the CORS configuration is applied as a public resource.
+
+| `support_credentials`
+| Optional
+| `false`
+| A boolean indicating if the CORS configuration supports credentials, such as cookies, authorization headers, and TLS client certificates.
+
+|===
+
+[[origin-group-methods]]
+=== cors::OriginGroup Struct Configuration Parameters
+
+[%header%autowidth.spread,cols="a,a,a,a"]
+|===
+|Property | Required or Optional | Default Value | Description
+
+| `plain_origins`
+| Optional
+| Empty `Vec`
+| `Vec` of origins included in the group, for example, `\http://www.the-origin-of-time.com`.
```

Comment:
```suggestion
| `Vec` of origins included in the group. For example, `+http://www.the-origin-of-time.com+`.
```
This is the syntax to prevent the build from creating a link from a URL. Not sure if escaping with \ works, it might be the case. Please confirm locally.